src/joystream-lib/api.ts

```diff
@@ -314,9 +314,13 @@ export class JoystreamJs {
       throw AccountNotFoundError
     }
 
-    const accountInjector = await web3FromAddress(accountId)
-    this._selectedAccountId = accountId
-    this.api.setSigner(accountInjector.signer)
+    try {
+      const accountInjector = await web3FromAddress(accountId)
+      this._selectedAccountId = accountId
+      this.api.setSigner(accountInjector.signer)
+    } catch (error) {
+      throw AccountNotFoundError
+    }
```

Comment:
Did you run into some case where this was needed? I'm curious why the previous check (`find`) wouldn't catch this

Reply:
The app was breaking when the `accountId` saved in `localStorage` was invalid.
![image](https://user-images.githubusercontent.com/51168865/115189671-23983800-a0e7-11eb-9390-1188b35f8e51.png)
Not super sure if it's still needed. The previous check, for some reason, is not catching this.

Reply:
oh, I see. The previous check doesn't do anything 😅  
```js
   if (!this._accounts.find((a) => a.id)) {
      throw AccountNotFoundError
    }
```
Should be: 
```js
    if (!this._accounts.find((a) => a.id === accountId)) {
      throw AccountNotFoundError
    }
```